app/models/merchant.rb

```diff
@@ -4,4 +4,9 @@ class Merchant < ApplicationRecord
   has_many :transactions, through: :invoices
 
   validates :name, presence: true
+
+  def self.get_favorite_customer(merchant)
+    Customer.find(Customer.joins(:transactions, :invoices).where("invoices.merchant_id = ? and result = ?", merchant.id, "success").select("customers.*, invoices.customer_id, transactions.result").group("invoices.customer_id").order("count_id DESC").count('id').first.first)
```

Comment:
@s-espinosa. I got it to work, but I had to find the customer from the query. I know that there's a way to just return the customer object, and I feel like I'm only one step off from that, but I can't get there. Do you have any advice on how to break it up into separate methods?